interface/billing/billing_process.php

```diff
@@ -235,7 +226,8 @@ function process_form($ar)
                     $bill_info[] = xl("Claim ") . $claimid . xl(" has been re-opened.") . "\n";
                 } elseif (isset($ar['bn_x12']) || isset($ar['bn_x12_encounter'])) {
                     $log = '';
-                    $segs = explode("~\n", gen_x12_837($patient_id, $encounter, $log, isset($ar['bn_x12_encounter'])));
+                    $x12 = new X12837P;
+                    $segs = explode("~\n", $x12->gen_x12_837($patient_id, $encounter, $log, isset($ar['bn_x12_encounter'])));
```

Comment:
This is going to be a static function (ie. mean do not need to initiate a class container since it is not holding anything).
So can:
1. remove the $x12 = new X12837P; line
2. and then do this:
```
$segs = explode("~\n", X12837P::gen_x12_837($patient_id, $encounter, $log, isset($ar['bn_x12_encounter'])));
```

Reply:
thanks for all of the detailed advice, i went looking for that modernizing codebase advice you mentioned to matthew a couple weeks ago on the call and figured i'd take a closer look at the code that i'm the most familiar with and i've found some items that will need fixing :)

Reply:
This is nice work to modernize (an end goal is to have no `require` commands at the top of each script and instead have `use` commands, which this gets up one step towards :) ).